__tests__/service/onesignal/OneSignalUtil.test.ts

```diff
@@ -22,17 +22,21 @@ beforeEach(() => {
 });
 
 describe('TESTING OneSignalUtil', () => {
-  it('TEST should call add tag and remove tag when calling resubscribe', async () => {
-    // Setup
-    getSubscribeableTopic.mockImplementationOnce(() => mockSubscribeableTopicResponse);
+  it(
+    'TEST should call add tag and remove tag when calling resubscribe',
+    async () => {
+      // Setup
+      getSubscribeableTopic.mockImplementationOnce(() => mockSubscribeableTopicResponse);
 
-    // Execution
-    await OneSignalUtil.rebuildAndSubscribeTags();
+      // Execution
+      await OneSignalUtil.rebuildAndSubscribeTags();
 
-    // Assertion
-    expect(OneSignal.User.removeTag).toHaveBeenCalledTimes(3);
-    expect(OneSignal.User.addTag).toHaveBeenCalledTimes(2);
-  });
+      // Assertion
+      expect(OneSignal.User.removeTag).toHaveBeenCalledTimes(3);
+      expect(OneSignal.User.addTag).toHaveBeenCalledTimes(2);
+    },
+    11 * 1000
+  );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The timeout for the test 'TEST should call add tag and remove tag when calling resubscribe' has been increased to 11 seconds (11000 milliseconds). This change is likely due to the increase in timeout between removing and registering OneSignal tags. However, it's important to ensure that this doesn't lead to slower test execution times overall. If there are many tests with similar timeouts, it might be worth considering ways to reduce the need for such long timeouts, such as by mocking the delay or using fake timers.
```diff
-    11 * 1000
+    jest.useFakeTimers();
+    await OneSignalUtil.rebuildAndSubscribeTags();
+    jest.runAllTimers();
```
This change replaces the real timer with Jest's fake timer, allowing the test to execute immediately without waiting for the actual delay.


<!-- This is an auto-generated comment by OSS CodeRabbit -->